presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/AbstractSqlInvokedFunctionNamespaceManager.java

```diff
@@ -211,11 +211,12 @@ public final ScalarFunctionImplementation getScalarFunctionImplementation(Functi
     }
 
     @Override
-    public CompletableFuture<Block> executeFunction(FunctionHandle functionHandle, Page input, List<Integer> channels, TypeManager typeManager)
+    public CompletableFuture<SqlFunctionResult> executeFunction(String source, FunctionHandle functionHandle, Page input, List<Integer> channels, TypeManager typeManager)
```

Comment:
Why not keep this as TaskId taskId rather than String source? TaskId already has a thrift representation, and its more descriptive of the meaning compared to "source"